### DISCUSSION TITLE
Support user token creation from the API

### DISCUSSION THREAD

**Discussion by TwistByrn** [nautobot/nautobot#471] [Ideas]
Would it be considered to allow the creation of a user token via the API?

**Comment by lampwins** [comment]
Yeah, this is totally something we want to do. We have had some off-the-cuff discussions about the best way to do it. This was also a conversation in netbox: https://github.com/netbox-community/netbox/issues/5264

Here is my contribution to that discussion:

> I see two paths ways here and maybe it makes sense to actually do both.
> 1. Users are able to authenticate to a new API endpoint /api/users/<username>/tokens/ to generate a new token. A POST request would create a new token, a GET request would show all the user's tokens. The key here is that a user does this for themselves, not anyone else. The endpoint would accept basic auth and use whatever auth backend to actually authenticate the user.
> 2. Admin users (those with permission to create/modify users) are able to generate and view tokens for other users.

**Reply by smk4664** [reply]
Number 1 would be really helpful for interacting with Nautobot.  I don't know if Number 2 would be useful, and looks like it goes against issue #901

**Comment by haresotto** [comment]
Hello,
I can confirm that this would be very helpful. In my case, I am using ansible to install and configure nautobot.
Unfortunately, I can't do this in a row because I need to wait for the installation to finish, create a token, set it in ansible then launch the configuration.
I would be glad to see this implemented :-).

**Comment by bryanculver** [comment]
This was implemented in 1.3.0: https://github.com/nautobot/nautobot/issues/1374

**Comment by haresotto** [comment]
@bryanculver @jeremystretch thank you (and everyone else involved) very much, this is appreciated :-).
